arch/arm/core/aarch32/fatal.c

```diff
@@ -83,9 +83,9 @@ void z_do_kernel_oops(const z_arch_esf_t *esf)
 	unsigned int reason = esf->basic.r0;
 
 #if defined(CONFIG_USERSPACE)
-	if ((__get_CONTROL() & CONTROL_nPRIV_Msk) == CONTROL_nPRIV_Msk) {
+	if (aarch32_is_unprivileged_state()) {
```

Comment:
can we specify in the name that we are talking about thread mode being nPRIV?
Cause, otherwise we have the arch_is_user_context API, but that one checks the handler mode as well..

Reply:
E.g. aarch32_thread_is_user_mode(), or sthg similar

Reply:
fixed